packages/compass-e2e-tests/tests/collection-ai-query.test.ts

```diff
@@ -0,0 +1,164 @@
+      getRequests: _getRequests,
+      clearRequests: _clearRequests,
+      setMockAtlasServerResponse: _setMockAtlasServerResponse,
+      stop,
+    } = await startMockAtlasServiceServer();
+
+    stopMockAtlasServer = stop;
+    getRequests = _getRequests;
+    clearRequests = _clearRequests;
+    setMockAtlasServerResponse = _setMockAtlasServerResponse;
+
+    process.env.COMPASS_ATLAS_SERVICE_BASE_URL_OVERRIDE = endpoint;
+
+    telemetry = await startTelemetryServer();
+    compass = await beforeTests({
+      extraSpawnArgs: ['--enableAIExperience'],
+    });
+    browser = compass.browser;
+  });
+
+  beforeEach(async function () {
+    await createNumbersCollection();
+    await browser.connectWithConnectionString();
+    await browser.navigateToCollectionTab('test', 'numbers', 'Documents');
+  });
+
+  after(async function () {
+    await stopMockAtlasServer();
+
+    delete process.env.COMPASS_ATLAS_SERVICE_BASE_URL_OVERRIDE;
+    delete process.env.COMPASS_E2E_SKIP_ATLAS_SIGNIN;
+
+    await afterTests(compass, this.currentTest);
+    await telemetry.stop();
+  });
+
+  afterEach(async function () {
+    clearRequests();
+    await afterTest(compass, this.currentTest);
+  });
+
+  describe('when the ai model response is valid', function () {
+    beforeEach(function () {
+      setMockAtlasServerResponse({
+        status: 200,
+        body: {
+          content: {
+            query: {
+              filter: {
+                i: {
+                  $gt: 50,
+                },
+              },
+            },
+          },
+        },
+      });
+    });
+
+    it('makes request to the server and updates the query bar with the response', async function () {
+      // Click the ask ai button.
+      await browser.clickVisible(Selectors.QueryBarAskAIButton);
+
+      // Enter the ai prompt.
+      await browser.clickVisible(Selectors.QueryBarAITextInput);
+
+      const testUserInput = 'find all documents where i is greater than 50';
+      await browser.setOrClearValue(
+        Selectors.QueryBarAITextInput,
+        testUserInput
+      );
+
+      // Click generate.
+      await browser.clickVisible(Selectors.QueryBarAIGenerateQueryButton);
+
+      // Wait for the ipc events to succeed.
+      await browser.waitUntil(async function () {
+        // Make sure the query bar was updated.
+        const queryBarFilterContent = await browser.getCodemirrorEditorText(
+          Selectors.queryBarOptionInputFilter('Documents')
+        );
+        return queryBarFilterContent === '{i: {$gt: 50}}';
+      });
+
+      // Check that the request was made with the correct parameters.
+      const requests = getRequests();
+      expect(requests.length).to.equal(1);
+      expect(requests[0].content.userInput).to.equal(testUserInput);
+      expect(requests[0].content.collectionName).to.equal('numbers');
+      expect(requests[0].content.databaseName).to.equal('test');
+      expect(requests[0].content.schema).to.exist;
```

Comment:
Feel free to keep it, but this is something we already cover with functional tests and I don't think redundancy is needed here. From our e2e tests I would mostly expect testing that the UI is correctly handling user interactions

I know that we do a similar thing for analytics, but this is not something we usually cover with unit / functional, so there it makes more sense to check specifically what got pushed to the backend

Reply:
Makes sense, I'll remove those extra checks in another pr to avoid another test ci run.